app/src/ui/history/commit-list.tsx

```diff
@@ -272,16 +275,26 @@ export class CommitList extends React.Component<ICommitListProps, {}> {
 
     const selectedShas = sorted.map(r => this.props.commitSHAs[r])
     const selectedCommits = this.lookupCommits(selectedShas)
-    this.props.onCommitsSelected(selectedCommits)
+    const indexes = selectedCommits.map(v =>
+      this.props.commitSHAs.findIndex(sha => sha === v.sha)
+    )
+    const selectionContainsInitialCommit =
+      this.getLastRetainedCommitRef(indexes) === null
```

Comment:
Is this really doing what we expect it to? Looking at the `getLastRetainedCommitRef` it seems to look at the last commit in the `commitSHAs` list is among the selected ones. That doesn't _necessary_ mean that the first commit of the branch is selected since `commitSHAs` is loaded incrementally.

So if you go to the history tab and hit <kbd>Cmd</kbd>+<kbd>A</kbd> to select all commits the `selectionContainsInitialCommit` will be true and we will diff against the null tree without having selected the initial commit.

I think what we'd want to do here is something like

```suggestion
    const selectionContainsInitialCommit =
      selectedCommits.at(-1)?.parentSHAs.length === 0

Reply:
Rendered moot since changed approach to try, check for bad revision, and then retry with null tree sha.